Controller/Integration/Products.php

```diff
@@ -0,0 +1,135 @@
+<?php declare(strict_types=1);
+
+namespace Elgentos\PrismicIO\Controller\Integration;
+
+use Magento\Catalog\Model\ResourceModel\Product\CollectionFactory;
+use Magento\Catalog\Ui\DataProvider\Product\ProductCollectionFactory;
+use Magento\Framework\App\Action\HttpGetActionInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\App\Request\Http;
+use Magento\Framework\Serialize\Serializer\Json;
+
+class Products implements HttpGetActionInterface
+{
+    /**
+     * @var Json
+     */
+    public $json;
+    /**
+     * @var ScopeConfigInterface
+     */
+    public $config;
+    /**
+     * @var JsonFactory
+     */
+    protected $jsonFactory;
+    /**
+     * @var CollectionFactory
+     */
+    protected $productCollectionFactory;
+    /**
+     * @var Http
+     */
+    protected $request;
+
+    /**
+     * Constructor
+     *
+     * @param Http $request
+     * @param JsonFactory $jsonFactory
+     * @param CollectionFactory $productCollectionFactory
+     * @param Json $json
+     * @param ScopeConfigInterface $config
+     */
+    public function __construct(
+        Http $request,
+        JsonFactory $jsonFactory,
+        CollectionFactory $productCollectionFactory,
+        Json $json,
+        ScopeConfigInterface $config,
+        \Magento\Store\Model\StoreManagerInterface $storeManager
+    )
+    {
+        $this->productCollectionFactory = $productCollectionFactory;
+        $this->jsonFactory = $jsonFactory;
+        $this->request = $request;
+        $this->json = $json;
+        $this->config = $config;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * Execute view action
+     *
+     * @return ResultInterface
+     */
+    public function execute()
+    {
+        $this->protectRoute();
+        // Make sure the default attributes needed for Prismic are added
+        $attributes = array_unique(
+            array_merge(
+                ['name', 'image', 'short_description', 'status', 'updated_at'],
+                explode(',', $this->config->getValue('prismicio/integration_fields/attributes'))
```

Comment:
Why use a separate configurationinterface? I always find that pattern annoying, now I have to edit 3 files to add a config option, with no real upside?

Reply:
Because currently the per store feature is not usable, which is something I can make required when defining it in my interface.. Whilst in your `sytem.xml` you define it can be used by store for this implementation.

Also, my single point of truth(my interface) without bothering the implementation(my model) is gone...

Same goes for using the internal API for products, but I didn't mention it yet

Reply:
Ok, good explanation. It's done.